Library/Homebrew/test/caveats_spec.rb

```diff
@@ -26,4 +26,53 @@ def caveats
       expect(described_class.new(f)).not_to be_empty
     end
   end
+
+  describe "#caveats" do
+    context "when f.plist is not nil" do
+      it "prints plist startup information when f.plist_startup is not nil" do
+        p = formula do
+              url "foo-1.0"
+              def plist
+                "plist_test.plist"
+              end
+              plist_options :startup => true
+            end
+        expect(described_class.new(p).caveats).to include("startup:\n  sudo brew")
+      end
+
+      it "prints plist login information when f.plist_startup is nil" do
+        p = formula do
+              url "foo-1.0"
+              def plist
+                "plist_test.plist"
+              end
+            end
+        expect(described_class.new(p).caveats).to include("login:\n  brew")
+      end
+    end
+
+    context "when f.keg_only is not nil" do
+      it "gives information about command to be run when f.bin is a directory" do
+        p = formula do
+              url "foo-1.0"
+              keg_only "some reason"
+            end
+        p_inst = described_class.new(p)
+        allow(p.bin).to receive(:directory?).and_return(true)
+        expect(p_inst.caveats).to include("This formula is keg-only")
```

Comment:
`caveats = described_class.new(f).caveats` above this line to and then you don't need to repeat `p_inst.caveats` or declare `p_inst`